src/components/CartOrderForm.js

```diff
@@ -85,7 +90,7 @@ export const CartOrderForm = ({
   };
 
   const isElegibleForFreeDelivery = () => {
-    return hasDeliveryDiscount || (cartTotal > amountForFreeDelivery && deliveryMethod === 'delivery');
+    return (hasDeliveryDiscount && !haveExtraFee) || (cartTotal > amountForFreeDelivery && deliveryMethod === 'delivery' && !haveExtraFee);
```

Comment:
would this be the same? looks like we're repeating the `&& !haveExtraFee` rule

```suggestion
    return (hasDeliveryDiscount || (cartTotal > amountForFreeDelivery && deliveryMethod === 'delivery')) && !haveExtraFee;
```

Reply:
Yes you are right.